ingester/ingester.go

```diff
@@ -515,7 +515,7 @@ func (i *Ingester) flushSeries(u *userState, fp model.Fingerprint, series *memor
 	}
 
 	firstTime := series.chunkDescs[0].FirstTime()
-	flush := immediate || model.Now().Sub(firstTime) > i.cfg.MaxChunkAge
+	flush := immediate || len(series.chunkDescs) > 1 || model.Now().Sub(firstTime) > i.cfg.MaxChunkAge
```

Comment:
Why _strictly_ greater than one? I could understand `> 0`, but `> 1` seems a bit odd. I realise at this point we know that it's non-empty, but why do we not want to flush if it has only 1 thing? 

Somewhat out of scope for this PR, but it's difficult for me to reason locally about `fpLocker`. Any reader of this method needs to know about its call-site to understand why locking is necessary.